### PR TITLE
More pipeline changes

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Move the other packaging pipelines to VS2019
2. Remove the build option: --use_eigenthreadpool. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml [hunk 1]
```diff
@@ -161,49 +158,78 @@ jobs:
...
-        EnvSetupScript: 'setup_env.bat'
-        buildArch: 'amd64'  # amd64 is needed for vcvars target arch
-        setVcvars: true
 
-    - task: PowerShell@1
-      displayName: 'Set CUDA path'
+    - task: PythonScript@0
+      displayName: 'Generate cmake config'
       inputs:
-        scriptName: 'tools/ci_build/github/windows/set_cuda_path.ps1'
```
Comment (jignparm) [inline tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: This was previously to allow different versions of CUDA builds. Does this mean only CUDA 10 will be supported going forward? If a build for CUDA 9 is required, can that still be done easily?

File: tools/ci_build/github/azure-pipelines/nuget/cpu-mklml-pipeline.yml [hunk 2]
```diff
@@ -10,5 +9,5 @@ variables:
 jobs: 
 - template: templates/cpu-mklml.yml
   parameters:
-    AgentPool : $(AgentPoolWin)
+    AgentPool : 'Win-CPU-2019'
```
Comment (jignparm) [inline tools/ci_build/github/azure-pipelines/nuget/cpu-mklml-pipeline.yml]: Does this agentpool exist on AIInfra vso instance (for ESRP signing)?

File: tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/c-api-packaging-pipelines.yml]: If we need to support a new version, just change the build args from "--cuda_version=$(CUDA_VERSION) --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v$(CUDA_VERSION) --cudnn_home=xxx" to something new. 

>  If a build for CUDA 9 is required, can that still be done easily?
I'm not sure if our code still compiles. Anyway, it should be very easy to support 10.1 and 10.2. 

The 'set_cuda_path.ps1' script was there because we didn't 'install' CUDA on the machines, we only copies it.  Now I put it into the VM image.  So it's not needed. We don't need to run it every time for every build.  And it's better to not change the system software during the build.

However, I haven't installed CUDA 10.1 and 10.2 on these machines yet.  It's easily doable.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/Microsoft.ML.OnnxRuntime.Gpu.EndToEndTests.RunCapi.vcxproj [hunk 3]
```diff
@@ -26,13 +24,13 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v141</PlatformToolset>
+    <PlatformToolset>v142</PlatformToolset>
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/Microsoft.ML.OnnxRuntime.Gpu.EndToEndTests.RunCapi.vcxproj]: Why the change to toolset v142? It is a requirement now?

File: tools/ci_build/github/azure-pipelines/nuget/cpu-mklml-pipeline.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/nuget/cpu-mklml-pipeline.yml]: YES

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/Microsoft.ML.OnnxRuntime.Gpu.EndToEndTests.RunCapi.vcxproj (hunk 3 above)
Comment (snnn) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/Microsoft.ML.OnnxRuntime.Gpu.EndToEndTests.RunCapi.vcxproj]: The build complains Windows SDK 8.1 is not found.  The new build machine doesn't have Windows SDK for 8.1.

Comment (jignparm) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/Microsoft.ML.OnnxRuntime.Gpu.EndToEndTests.RunCapi.vcxproj]: Ok -- users will require same toolset to run the test projects, but that should be fine.
